ci/kind/test-conformance-kind.sh

```diff
@@ -0,0 +1,187 @@
+#!/usr/bin/env bash
+
+# Copyright 2023 Antrea Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# The script runs kind e2e tests with different traffic encapsulation modes.
```

Comment:
Updated the file name to be more specific. I checked with @antoninbas and think it might make sense to use a separate file since the kind setup id already in another file, and we do not need most of the options in `test-e2e-kind.sh` and might need different test strategies in the future.